src/cryptonote_core/blockchain.cpp

```diff
@@ -4845,7 +4844,7 @@ bool Blockchain::prepare_handle_incoming_blocks(const std::vector<block_complete
     {
       m_blocks_longhash_table.clear();
       uint64_t thread_height = height;
-      tools::threadpool::waiter waiter;
+      tools::threadpool::waiter waiter(tpool);
```

Comment:
There doesn't seem to be an effective method to see if this section failed via exception.